weave-js/src/core/ops/primitives/date.ts

```diff
@@ -36,6 +36,98 @@ export const opDateSub = makeDateOp({
   },
 });
 
+export const opDateAddDuration = makeDateOp({
+  hidden: true,
+  name: 'date-addDuration',
+  argTypes: {date: 'date', duration: 'number'},
+  renderInfo: {
+    type: 'binary',
+    repr: '+',
+  },
+  description: `Returns the sum between a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  argDescriptions: {
+    date: `Date ${docType('date')}`,
+    duration: `Duration ${docType('number')}`,
+  },
+  returnValueDescription: `The sum between the a ${docType(
+    'date'
+  )} and duration in milliseconds ${docType('number')}`,
+  returnType: inputTypes => 'date',
+  resolver: inputs => {
+    if (
+      inputs.date == null ||
+      inputs.duration == null ||
+      inputs.duration <= 0
+    ) {
+      return null;
+    }
+    const dateAddDuration = moment
+      .utc(inputs.date)
+      .add(moment.duration(inputs.duration, 'milliseconds'))
+      .toDate();
+    return new Date(dateAddDuration);
+  },
+});
+
+export const opDateDiffDaysStringFormat = makeDateOp({
+  hidden: true,
+  name: 'date-diffDaysStringFormat',
+  argTypes: {lhs: 'date', rhs: 'date'},
+  renderInfo: {
+    type: 'binary',
+    repr: '-',
+  },
+  description: `Returns the difference between two ${docType('date', {
+    plural: true,
+  })} in days in a nice string form`,
+  argDescriptions: {
+    lhs: `First ${docType('date')}`,
+    rhs: `Second ${docType('date')}`,
+  },
+  returnValueDescription: `The difference between the two ${docType('date', {
+    plural: true,
+  })} in days in a nice string form`,
+  returnType: inputTypes => 'string',
+  resolver: inputs => {
+    if (inputs.lhs == null || inputs.rhs == null) {
+      return null;
+    }
+    const dateDiff = moment
+      .utc(inputs.lhs)
+      .diff(moment.utc(inputs.rhs), 'milliseconds');
+    const dayDiff = Math.floor(
+      moment.duration(dateDiff, 'milliseconds').asDays()
+    );
+    if (dayDiff <= 0) {
+      return 'less than 1 day';
+    }
+    return dayDiff.toString() + (dayDiff === 1 ? ' day' : ' days');
+  },
+});
```

Comment:
Right now this function is pretty specific to this ttl use case, but there are other places we might want to use this and by deleting some code we can make it more general. Can you make it so that:

1. This function uses moment.to() to calculate the string diff https://momentjs.com/docs/#/displaying/to/
2. This takes two timestamps instead of dates
3. By default this returns whatever the default nice representation of the date diff is from moment.to()
4. The op takes some config parameters to round to the nearest whatever (in your case you would pass config parameters to round to the nearest day)


Here are some test cases I would expect to pass with the default parameters:

0 to 44 seconds | s | in seconds
45 to 89 seconds | m | in a minute
90 seconds to 44 minutes | mm | in 2 minutes ... in 44 minutes
45 to 89 minutes | h | in an hour
90 minutes to 21 hours | hh | in 2 hours ... in 21 hours
22 to 35 hours | d | in a day
36 hours to 25 days | dd | in 2 days ... in 25 days
26 to 45 days | M | in a month
45 to 319 days | MM | in 2 months ... in 10 months
320 to 547 days (1.5 years) | y | in a year
548 days+ | yy | in 2 years ... in 20 years

And then if the config is passsed with round to the nearest day, then we round and show the nice time in units of days.